pkg/reconciler/cronjobsource/cronjobsource.go

```diff
@@ -157,13 +150,25 @@ func (r *Reconciler) reconcile(ctx context.Context, cronjob *v1alpha1.CronJobSou
 	}
 	cronjob.Status.MarkSink(sinkURI)
 
+	_, err = cron.ParseStandard(cronjob.Spec.Schedule)
+	if err != nil {
+		cronjob.Status.MarkInvalidSchedule("Invalid", "")
+		return fmt.Errorf("invalid schedule: %v", err)
+	}
+	cronjob.Status.MarkSchedule()
+
 	ra, err := r.createReceiveAdapter(ctx, cronjob, sinkURI)
 	if err != nil {
-		logging.FromContext(ctx).Error("Unable to create the receive adapter", zap.Error(err))
+		r.Logger.Error("Unable to create the receive adapter", zap.Error(err))
 		return fmt.Errorf("creating receive adapter: %v", err)
 	}
 	cronjob.Status.PropagateDeploymentAvailability(ra)
 
+	track := r.resourceTracker.TrackInNamespace(cronjob)
+	if err = track(utils.ObjectRef(ra, deploymentGVK)); err != nil {
```

Comment:
Same as with ApiServerSource, due to #1998, don't use this tracker, instead have controller.go setup the informer.